### PR TITLE
Graphs working

### LINKED ISSUE #1397: Add Linode graph - traffic ipv4
Implement linode traffic ipv4 graph

### LINKED ISSUE #1398: Add linode graph - ipv6 traffic

### LINKED ISSUE #1399: Add Linode graph - IO

### LINKED ISSUE #1400: Add Linode graph - CPU

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #1400, closes #1399, closes #1398, closes #1397

screen shot 2017-03-23 at 4 03 41 pm
screen shot 2017-03-23 at 4 03 21 pm
screen shot 2017-03-23 at 4 04 07 pm
screen shot 2017-03-23 at 4 04 20 pm

### CODE REVIEW

File: src/linodes/linode/layouts/DashboardPage.js [hunk 1]
```diff
@@ -11,135 +11,136 @@ import { getLinode } from './IndexPage';
...
 import { Card } from '~/components/cards';
+import LineGraph from '~/components/graphs/LineGraph';
+import { Select } from '~/components/form';
+import { objectFromMapByLabel } from '~/api/util';
+import { linodeStats } from '~/api/linodes';
+
+function formatData(datasets) {
+  return {
+    labels: datasets[0].map(datum => datum[0]),
+    datasets: datasets.map(dataset => ({
```
Comment (mparke) [inline src/linodes/linode/layouts/DashboardPage.js]: would you mind putting a comment in here re: what these objects look like?

Comment (mparke) [inline src/linodes/linode/layouts/DashboardPage.js]: or maybe..below on the specific types, now that I see them

**Comment by mparke** [comment]
missing the chart dep?

**Comment by eatonphil** [comment]
Oh yeah, it went into another pr by mistake.

**Comment by mparke** [comment]
I like the color palette find, but any thoughts on excluding red & pink in this context? Red appears pretty alarming here

**Comment by eatonphil** [comment]
Removed red and pink:

screen shot 2017-03-27 at 3 28 25 pm

File: src/components/graphs/LineGraph.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      this._chart.render();
+    } else {
+      this._chart = new Chart(ctx, config);
+    }
+  }
+
+  render() {
+    return (
+      <canvas
+        style={{ marginTop: '15px', width: '100%', height: '500px', maxHeight: '500px' }}
```
Comment (mparke) [inline src/components/graphs/LineGraph.js]: inline styles? can any of these be moved into scss?

Comment (eatonphil) [inline src/components/graphs/LineGraph.js]: Yep, forgot about this.

File: src/components/graphs/LineGraph.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import Chart from 'chart.js';
+import moment from 'moment';
+import _ from 'lodash';
+
+// Source: http://stackoverflow.com/a/5624139/1507139
+function rgbaFromHex(hex, alpha) {
```
Comment (mparke) [inline src/components/graphs/LineGraph.js]: said was not needed?

**Comment by mparke** [comment]
Tooltip text renders blurry? also these numbers are huge ints, and don't match the Y axis

screen shot 2017-03-28 at 10 32 31 am

**Comment by mparke** [comment]
Should cpu Y-axis be scaled larger? it's not really useful to see a scale this small with large spikes?

screen shot 2017-03-28 at 10 32 58 am

**Comment by mparke** [comment]
Negative Y axis values?

screen shot 2017-03-28 at 10 33 14 am
screen shot 2017-03-28 at 10 33 20 am

**Comment by eatonphil** [comment]
@mparke made some issues to deal with what you're seeing:

> Tooltip text renders blurry?

See https://github.com/linode/manager/issues/1470

>  also what is the huge int here?

Unformatted date seconds. See https://github.com/linode/manager/issues/1469

> Should cpu Y-axis be scaled larger? it's not really useful to see a scale this small with large spikes?

See https://github.com/linode/manager/issues/1468

> Negative Y axis values?

See https://github.com/linode/manager/issues/1467

**Comment by abemassry** [comment]
When trying to create a new linode

Uncaught (in promise) TypeError: Cannot read property 'cpu' of undefined DashboardPage.js?3d84:54

**Comment by eatonphil** [comment]
@abemassry addressed, committed, and pushed.

**Comment by eatonphil** [comment]
Thank you!

**Pull request merged**